src/neo/Network/P2P/TaskManager.cs

```diff
@@ -271,6 +271,11 @@ private void RequestTasks(TaskSession session)
             else if (Blockchain.Singleton.HeaderHeight >= session.LastBlockIndex
                     && TimeProvider.Current.UtcNow.ToTimestamp() - PingCoolingOffPeriod >= Blockchain.Singleton.GetBlock(Blockchain.Singleton.CurrentHeaderHash)?.Timestamp)
             {
+                if (session.AvailableTasks.Remove(MemPoolTaskHash))
+                {
+                    session.RemoteNode.Tell(Message.Create(MessageCommand.Mempool));
+                }
+
                 session.RemoteNode.Tell(Message.Create(MessageCommand.Ping, PingPayload.Create(Blockchain.Singleton.Height)));
             }
```

Comment:
Or put the condition behind the `else if` block,

```cs
if (session.AvailableTasks.Remove(MemPoolTaskHash))
{
      session.RemoteNode.Tell(Message.Create(MessageCommand.Mempool));
}
```

Reply:
![image](https://user-images.githubusercontent.com/3167973/82329703-01cb9800-99e2-11ea-8a2c-6f86e113d380.png)

But this condition already exists, maybe I don't understand you ..